server/routes/induction/create/wantToAddQualificationsCreateController.test.ts

```diff
@@ -0,0 +1,141 @@
+    },
+  } as InductionDto
+
+  let req: Request
+  const res = {
+    redirect: jest.fn(),
+    render: jest.fn(),
+  } as unknown as Response
+  const next = jest.fn()
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    req = {
+      session: { prisonerSummary },
+      body: {},
+      user: { token: 'some-token' },
+      params: { prisonNumber },
+      flash: jest.fn(),
+      path: `/prisoners/${prisonNumber}/create-induction/want-to-add-qualifications`,
+    } as unknown as Request
+  })
+
+  describe('getWantToAddQualificationsView', () => {
+    it('should get the Want To Add Qualifications view', async () => {
+      // Given
+      req.session.inductionDto = partialInductionDto
+
+      const functionalSkills = validFunctionalSkills()
+      req.session.prisonerFunctionalSkills = functionalSkills
+      req.session.wantToAddQualificationsForm = undefined
+
+      const expectedWantToAddQualificationsForm: WantToAddQualificationsForm = {
+        wantToAddQualifications: undefined,
+      }
+
+      const expectedFunctionalSkills = functionalSkills
+      const expectedView = {
+        prisonerSummary,
+        backLinkUrl: '/prisoners/A1234BC/create-induction/reasons-not-to-get-work',
+        backLinkAriaText: `Back to What could stop Jimmy Lightfingers working when they are released?`,
+        form: expectedWantToAddQualificationsForm,
+        functionalSkills: expectedFunctionalSkills,
+        errors: noErrors,
+      }
+
+      // When
+      await controller.getWantToAddQualificationsView(req, res, next)
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith(
+        'pages/induction/prePrisonEducation/wantToAddQualifications',
+        expectedView,
+      )
+      expect(req.session.wantToAddQualificationsForm).toBeUndefined()
+    })
+  })
+
+  describe('submitWantToAddQualificationsForm', () => {
+    it('should not proceed to next page given form is submitted with validation errors', async () => {
+      // Given
+      req.session.inductionDto = partialInductionDto
+
+      const invalidWantToAddQualificationsForm = {
+        wantToAddQualifications: '',
+      }
+      req.body = invalidWantToAddQualificationsForm
+      req.session.wantToAddQualificationsForm = undefined
+
+      const expectedErrors = [
+        {
+          href: '#wantToAddQualifications',
+          text: `Select whether Jimmy Lightfingers wants to record any other educational qualifications`,
+        },
+      ]
+
+      // When
+      await controller.submitWantToAddQualificationsForm(req, res, next)
+
+      // Then
+      expect(res.redirect).toHaveBeenCalledWith('/prisoners/A1234BC/create-induction/want-to-add-qualifications')
+      expect(req.flash).toHaveBeenCalledWith('errors', expectedErrors)
+      expect(req.session.wantToAddQualificationsForm).toEqual(invalidWantToAddQualificationsForm)
+    })
+
+    it(`should proceed to qualification level page given user wants to add qualifications`, async () => {
```

Comment:
Is it worth having a unit test here for the other scenario? Where the user answers No?
In that case we would expect the redirect to be to the additional-training page